SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+
+    {
+        "notification": "We've succesfully received your request of registering
+        editions 1, 2, 5, 59 for piece 1234."
+        "websocket": "ws://ascribe.io/api/pieces/1234/editions"
+    }
+```
+
+In this case, a user of the API can use the `websocket` property of
+the response to listen in on the execution of the ongoing asynchronously
+executing task.
+Eventual mistakes of the API user's input, as for example submitting an edition
+number for an edition that has already been created in a previous request are
+all broadcasted only through this channel.
+Semantically wrong request content though, will only stop execution of the
+edition number in question, not the creation of all editions in the request.
+
+For example, assume the first request of this section went through successfully.
+Say, we make another request (btw. we do not need to wait for completion of the
+first request's tasks):
+
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "editions": [2]
+    }
+```
+
+Note, that this is the second time the API user requests the creation of
+edition 2 for piece 1234.
+Still, the API will respond with the above proposed schema (202 Accepted), but
+will eventually tell the API user via the provided websocket in the response that
+edition 2 could not have been registered, as it was already.
+
+
+### Freezing a set of unlimited editions
+
+Using `POST /api/pieces/1234/editions/`, submitting `num_editions` an API user
+is able to freeze/limit the unlimited editions of a piece.
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "num_editions": 59
+    }
+```
+
+Note that here `num_editions` needs to be at least greater than the greatest
+number used before when creating a set of unlimited editions.
+As we only registered the editions 1, 2, 5, 59 in previous example, this
+implies that when calling freeze on piece 1234, the server will create all
+remaining editions in the range of 2 - 58 (with the exception of 5).
+
+In turn this means, that the endpoint could be called with `num_editions` > 0 before
+calling it ever with `editions`, and would still create all editions in a range
+of 1 and value of `num_editions` (which is intended behavior!).
+
+Similarly to the response of creating unlimited editions, the response for
+freezing them returns:
+
+```javascript
+202 Accepted
+
+    {
+        "notification": "We've succesfully received your request of registering
+        editions 2, 3, 4, 6, ..., 58 for piece 1234."
+        "websocket": "ws://ascribe.io/api/pieces/1234/editions"
+    }
+```
+
+
+## Comments
+
+The old endpoints:
+
+    1. Set `num_editions` when registering a piece `POST /api/pieces`
+    2. Set `num_editions` and `piece_id` using `POST /api/editions`
+
+will stay existent and create a limited set of editions.
+They will however be removed from the public documentation, in order to phase
+them out in the long term, as the proposed endpoints are able to fulfill their tasks.
```

Comment:
We could perhaps just mark them with a deprecated notice, and agree on a date at which time they will be removed.